processor/src/main/java/org/mapstruct/ap/MappingProcessor.java

```diff
@@ -393,6 +406,26 @@ public TypeElement visitTypeAsClass(TypeElement e, Void p) {
         );
     }
 
+    /**
+     * Fetch the custom supported options provided by the SPI {@link CustomSupportedOptionsProvider}.
+     *
+     * @return the custom supported options
+     */
+    private static Set<String> getCustomSupportedOptions() {
+        final CustomSupportedOptionsProvider customSupportedOptionsProvider =
+            Services.get( CustomSupportedOptionsProvider.class, new DefaultCustomSupportedOptionsProvider() );
+        final Set<String> customSupportedOptions = customSupportedOptionsProvider.getCustomSupportedOptions();
+
+        // Ensure custom options are not in the mapstruct namespace
+        customSupportedOptions.forEach( option -> {
+            if (option.startsWith( "mapstruct" )) {
```

Comment:
is this enough to check that the custom option does not start with `mapstruct` or do we want to be more thorough?

Reply:
this is enough yes